javascript/date_service.js

```diff
@@ -12,4 +12,29 @@ FinancialFreedom.service('DateService', function() {
     this.addDaysToDate = function(date, days) {
         date.setDate(date.getDate() + days);
     };
+
+    this.calculateYearsBetween = function(date1, date2) {

```

Comment:
Let's talk through both these functions in person, I think it will be easier than trying to go over logic here.